modules/sql/src/test/java/IgniteSqlTest.java

```diff
@@ -0,0 +1,384 @@
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+/**
+ * Tests IgniteSQL facade API.
+ */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class IgniteSqlTest {
+    @Mock
+    IgniteSql queryMgr;
+    
+    @Mock
+    private IgniteTransactions igniteTx;
+    
+    @Mock
+    private Transaction tx;
+    
+    @BeforeEach
+    void setUp() {
+        initMock();
+    }
+    
+    @Test
+    public void testSyncSql() {
+        igniteTx.runInTransaction(tx -> {
+            Session sess = queryMgr.newSession();
+            
+            sess.defaultTimeout(10_000, TimeUnit.MILLISECONDS); // Set default timeout.
+            sess.property("memoryQuota", 10 * Constants.MiB); // Set default quota.
+            
+            // Execute outside TX.
+            ResultSet rs = sess.execute("INSERT INTO tbl VALUES (?, ?)", 10, "str");
+            
+            assertEquals(1, rs.updateCount());
+            
+            // Execute in TX.
+            Session txSession = tx.wrap(queryMgr.newSession());
+            
+            rs = txSession.execute("SELECT id, val FROM tbl WHERE id < {};", 10);
+            
+            for (SqlRow r : rs) {
+                assertTrue(10 > r.longValue("id"));
+                assertTrue((r.stringValue("val")).startsWith("str"));
+            }
+            
+            tx.commit();
+        });
+        
+        Mockito.verify(tx).commit();
+    }
+    
+    @Test
+    public void testSyncSql2() {
+        RecordView<Tuple> tbl = getTable();
+        
+        // Starts new TX.
+        Session txSession = tx.wrap(queryMgr.newSession());
+        
+        ResultSet rs = txSession.execute("SELECT id, val FROM tbl WHERE id < {};", 10);
+        SqlRow row = rs.iterator().next();
+        
+        tbl.withTransaction(tx)
+                .insertAsync(Tuple.create().set("val", "NewValue"))
+                .thenAccept(r -> txSession.transaction().rollback());
+        
+        Mockito.verify(tx, Mockito.times(1)).rollback();
+    }
+    
+    @Test
+    public void testSyncMultiStatementSql() {
+        Session sess = tx.wrap(queryMgr.newSession());
+        
+        MultiResultSet multiRs = sess.executeScript(
+                "CREATE TABLE tbl(id INTEGER PRIMARY KEY, val VARCHAR);"
+                        + "INSERT INTO tbl VALUES (1, 2);"
+                        + "SELECT id, val FROM tbl WHERE id == {};"
+                        + "DROP TABLE tbl", 10);
+        
+        Iterator<ResultSet> iterator = multiRs.iterator();
+        //TODO: Can iterator return null??
```

Comment:
Whan does `null` mean for `ResultSet`?

Reply:
As I remember we have an agreement about an impossibility of usage DDL and SQL statements in the same transaction. Need to discuss again.